### PR TITLE
Fixed #6517 -- Made dbshell use charset option on MySQL.

### PR DESCRIPTION
**Pull request by manav014**
[ticket-6517](https://code.djangoproject.com/ticket/6517)

I have added charset option in django\db\backends\mysql\client.py and it's respective tests in tests\dbshell\test_mysql.py

### CODE REVIEW

File: tests/dbshell/test_mysql.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def test_basic_params_specified_in_settings(self):
                 'HOST': 'somehost',
                 'PORT': 444,
                 'OPTIONS': {},
+                'DEFAULT-CHARACTER-SET': '',
```
Comment (felixxm) [inline tests/dbshell/test_mysql.py:22]: Please remove `DEFAULT-CHARACTER-SET` from all tests.

File: tests/dbshell/test_mysql.py [hunk 2]
```diff
@@ -112,3 +118,17 @@ def get_command_line_arguments(self, connection_settings, parameters=None):
...
+            ['mysql', '--user=someuser', '--password=somepassword',
+             '--host=somehost', '--port=444', 'somedbname', '--default-character-set=utf8'],
+            self.get_command_line_arguments({
+                'NAME': 'somedbname',
+                'USER': 'someuser',
+                'PASSWORD': 'somepassword',
+                'HOST': 'somehost',
+                'PORT': 444,
+                'OPTIONS': {},
+                'DEFAULT-CHARACTER-SET': 'utf8',
```
Comment (felixxm) [inline tests/dbshell/test_mysql.py:133]: We should add `charset` to `OPTIONS`:
```suggestion
                'OPTIONS': {
                    'charset': 'utf8',
                },
```

**Review by felixxm — COMMENTED** [review]
@manav014 Thanks for this patch :+1: I left comments.

File: tests/dbshell/test_mysql.py [hunk 3]
```diff
@@ -122,13 +116,14 @@ def get_command_line_arguments(self, connection_settings, parameters=None):
     def test_charset_configured(self):
         self.assertEqual(
             ['mysql', '--user=someuser', '--password=somepassword',
-             '--host=somehost', '--port=444', 'somedbname', '--default-character-set=utf8'],
```
Comment (felixxm) [inline tests/dbshell/test_mysql.py]: This is what we want to test. Please restore `'--default-character-set=utf8'` in this line.

File: django/db/backends/mysql/client.py [hunk 4]
```diff
@@ -21,6 +21,7 @@ def settings_to_cmd_args(cls, settings_dict, parameters):
         client_cert = settings_dict['OPTIONS'].get('ssl', {}).get('cert')
         client_key = settings_dict['OPTIONS'].get('ssl', {}).get('key')
         defaults_file = settings_dict['OPTIONS'].get('read_default_file')
+        charset = settings_dict['OPTIONS'].get('charset', settings_dict['DEFAULT-CHARACTER-SET'])
```
[the remark this answers is not in the document]
Comment (manav014) [reply django/db/backends/mysql/client.py:24]: Will remove it and improve the patch. Update you soon with this. Thank You!

**Comment by felixxm** [comment]
@manav014 I'm not really sure why you remove the fix and added test that does not test this patch :confused: I restored the fix and pushed edits.

**Comment by manav014** [comment]
> @manav014 I'm not really sure why you remove the fix and added test that does not test this patch :confused: I restored the fix and pushed edits. 

Actually I am working on it now and it is not completed yet. So just give me sometime to fix this. Sorry for the delay!🙂

**Comment by felixxm** [comment]
@manav014 Sorry, I assumed that you had completed updates. It's now polished and ready to merge. You don't need to update anything.

**Comment by manav014** [comment]
> @manav014 Sorry, I assumed that you had completed updates. It's now polished and ready to merge. You don't need to update anything.

Okay, Thank You! You may please merge the same.

**Pull request merged**